mailet/standard/src/main/java/org/apache/james/transport/matchers/RecipientDomainIs.java

```diff
@@ -0,0 +1,79 @@
+/****************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.transport.matchers;
+
+import java.util.Collection;
+
+import org.apache.james.core.Domain;
+import org.apache.james.core.MailAddress;
+import org.apache.mailet.base.GenericRecipientMatcher;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.github.steveash.guavate.Guavate;
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.base.Preconditions;
+import com.google.common.base.Splitter;
+
+/**
+ * This will return recipients mathcing a configured domain.
+ * <p>
+ * Sample configuration:
+ * </p>
+ * 
+ * <PRE>
+ * <CODE>
+ * &lt;mailet match=&quot;RecipientDomainIs=&lt;domain.com&gt;&quot; class=&quot;&lt;any-class&gt;&quot;&gt;
+ * </CODE>
+ * </PRE>
+ * @version CVS $Revision$ $Date$
+ * @since 3.2.0
+ */
+public class RecipientDomainIs extends GenericRecipientMatcher {
+    private static final Logger LOGGER = LoggerFactory.getLogger(RecipientDomainIs.class);
+
+    private Collection<Domain> receipientDomains;
+
+    @Override
+    public void init() {
+        String condition = getCondition();
+        Preconditions.checkNotNull(condition, "'condition' should not be null");
+
+        receipientDomains = parseDomainsList(condition);
+    }
+
+    @VisibleForTesting
+    Collection<Domain> parseDomainsList(String condition) {
+        return Splitter.onPattern("(, |,| )").omitEmptyStrings().splitToList(condition).stream().map(Domain::of)
```

Comment:
can you please add some carriage returns to help reading this please?
```suggestion
        return Splitter.onPattern("(, |,| )")
            .omitEmptyStrings()
            .splitToList(condition)
            .stream()
            .map(Domain::of)
```